### ISSUE TITLE
The mortar doesn't appear to respect NBT tags

### ISSUE THREAD

**Issue by belathus** [codetaylor/advancedmortars#33]
I wrote this script: https://github.com/belathus/Wanderlust-Renewed/blob/master/scripts/advancedmortars/thaumcraft.zs

It shows up in JEI correctly.

However, when testing it, the mortar consumed 3 of the same vis crystals, rather than 3 different vis crystals. I then tried to make salis mundis using only one crystal, and it worked. I'm guessing that it'd work with any vis crystal, which isn't what I want.

**Comment by codetaylor** [comment]
Reproduced with this:

```javascript
import mods.advancedmortars.Mortar;

val one = <minecraft:stick>.withTag({SomeTag: "one"});
val two = <minecraft:stick>.withTag({SomeTag: "two"});
val three = <minecraft:stick>.withTag({SomeTag: "three"});

Mortar.addRecipe(["diamond"], <minecraft:furnace>, 2, [one, two, three]);

recipes.addShapeless(one, [<minecraft:cobblestone>]);
recipes.addShapeless(two, [<minecraft:dirt>]);
recipes.addShapeless(three, [<minecraft:string>]);
```

**Comment by codetaylor** [comment]
Should be fixed in 1.12.2-1.6.21
